File: BE/src/main/java/com/example/hotel/model/entity/Room.java
package com.example.hotel.model.entity;

import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.time.LocalDateTime;
@Builder
@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Room {

    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Long id;

    private String name;

    private Long roomTypeId;

    private String description;

    private double area;

    private int size;

    @CreationTimestamp
    private LocalDateTime createdTime;

    @UpdateTimestamp
    private LocalDateTime updatedTime;
}

File: FE-ADMIN/demo/src/app/core/model/room.ts
export class Room {
  id: number;
  name: string;
  roomTypeId: number;
  description: string;
  area: number;
  size: number

  constructor() {
    this.id = 0
    this.name = "";
    this.roomTypeId = 0;
    this.description=  "";
    this.area = 0;
    this.size = 0;
  }

}
